web/src/components/select/index.tsx

```diff
@@ -53,7 +55,7 @@ export const Select: React.FC<Props> = ({
         {...getToggleButtonProps()}
         data-testid="select-toggle-btn">
         <span>
-          {selectedItem ? l10n.getString(selectedItem) : placeHolderText}
+          {selectedItem ? (doTranslation ? l10n.getString(selectedItem) : selectedItem) : placeHolderText}
```

Comment:
You can refactor this to use if/else. Nested ternaries are hard to understand

Reply:
It was in the JSX thou, I took the logic out for this.

Reply:
Done